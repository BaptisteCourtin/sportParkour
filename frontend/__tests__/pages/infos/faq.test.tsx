import "@testing-library/jest-dom";
import { render, screen, fireEvent } from "@testing-library/react";

import Faq from "@/pages/infos/faq";

// Mock des composants MUI
jest.mock(
  "@mui/material/Accordion",
  () =>
    ({ children }: { children: React.ReactNode }) =>
      <div data-testid="accordion">{children}</div>
);
jest.mock(
  "@mui/material/AccordionSummary",
  () =>
    ({ children }: { children: React.ReactNode }) =>
      <div data-testid="accordion-summary">{children}</div>
);
jest.mock(
  "@mui/material/AccordionDetails",
  () =>
    ({ children }: { children: React.ReactNode }) =>
      <div data-testid="accordion-details">{children}</div>
);
jest.mock(
  "@mui/material/AccordionActions",
  () =>
    ({ children }: { children: React.ReactNode }) =>
      <div data-testid="accordion-actions">{children}</div>
);
jest.mock(
  "@mui/material/Button",
  () =>
    ({
      children,
      onClick,
    }: {
      children: React.ReactNode;
      onClick?: () => void;
    }) =>
      <button onClick={onClick}>{children}</button>
);

// ---------------------------------------------------------------------------

describe("Faq Page", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Faq />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the title", () => {
    render(<Faq />);
    expect(
      screen.getByText("Questions fr√©quemment pos√©es")
    ).toBeInTheDocument();
  });

  it("renders three accordion items", () => {
    render(<Faq />);
    const accordions = screen.getAllByTestId("accordion");
    expect(accordions).toHaveLength(3);
  });

  it("renders correct questions", () => {
    render(<Faq />);
    expect(
      screen.getByText("Q : Comment vous contacter ?")
    ).toBeInTheDocument();
    expect(screen.getByText("Q : Vous aimez le sport ?")).toBeInTheDocument();
    expect(screen.getByText("Q : √áa va ?")).toBeInTheDocument();
  });

  it("renders correct answers", () => {
    render(<Faq />);
    expect(screen.getByText("Ne le fais pas.")).toBeInTheDocument();
    expect(screen.getByText("Non, mais l'argent oui.")).toBeInTheDocument();
    expect(screen.getByText("Moi ouais.")).toBeInTheDocument();
  });

  it("handles user interaction correctly", () => {
    render(<Faq />);
    const positiveButton = screen.getByText("Moi aussi √ßa va. üëç");
    const negativeButton = screen.getByText(
      "Moi non, mais tout le monde s'en fout."
    );

    fireEvent.click(positiveButton);
    expect(screen.getByText("C'est bien")).toBeInTheDocument();

    fireEvent.click(negativeButton);
    expect(screen.getByText("T'as raison")).toBeInTheDocument();
  });
});
