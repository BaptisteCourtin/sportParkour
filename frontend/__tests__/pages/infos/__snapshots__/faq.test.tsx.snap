// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Faq Page renders correctly 1`] = `
<DocumentFragment>
  <main
    class="pagesInfos"
  >
    <h1>
      Questions fr√©quemment pos√©es
    </h1>
    <div>
      <div
        data-testid="accordion"
      >
        <div
          data-testid="accordion-summary"
        >
          <h3>
            Q : Comment vous contacter ?
          </h3>
        </div>
        <div
          data-testid="accordion-details"
        >
          Ne le fait pas
        </div>
      </div>
      <div
        data-testid="accordion"
      >
        <div
          data-testid="accordion-summary"
        >
          <h3>
            Q : Vous aimez le sport ?
          </h3>
        </div>
        <div
          data-testid="accordion-details"
        >
          Non, mais l'argent oui.
        </div>
      </div>
      <div
        data-testid="accordion"
      >
        <div
          data-testid="accordion-summary"
        >
          <h3>
            Q : √ßa va ?
          </h3>
        </div>
        <div
          data-testid="accordion-details"
        >
          Moi ouais.
        </div>
        <div
          data-testid="accordion-actions"
        >
          <div>
            <button>
              moi aussi √ßa va üëç
            </button>
            <button>
              moi non, mais tout le monde s'en fout
            </button>
          </div>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
